packages/common/src/ui/layout/tables/DataTable.tsx

```diff
@@ -77,19 +77,17 @@ export const DataTableComponent = <T extends RecordWithId>({
     if (page * first > total) onChangePage(0);
   }, [pagination]);
 
-  if (isLoading)
+  if (isLoading) return <BasicSpinner />;
+
+  if (isError) {
     return (
-      <Box
-        sx={{
-          display: 'flex',
-          flex: 1,
-          justifyContent: 'center',
-          alignItems: 'center',
-        }}
-      >
-        <CircularProgress />
+      <Box sx={{ padding: 2 }}>
```

Comment:
passing the error down here, and letting the data table do the error display. much neater than repeating this on a bunch of pages